pennylane/qchem/factorization.py

```diff
@@ -274,44 +274,151 @@ def basis_rotation(one_electron, two_electron, tol_factor=1.0e-5):
+    in the spatial basis. During the tranformation, some extra one-body terms come out which are returned as well.
+
+    Args:
+        two_body_tensor (ndarray): a two-electron integral tensor giving the :math:`h_{pqrs}`.
+        spatial_basis (bool): True if the two-body terms are passed in spatial-orbital basis. False if they are in spin basis.
+
+    Returns:
+        tuple(ndarray, ndarray): a one-body :math:`a^\dagger_p a_q` term which come out during the processs and the transformed two-body tensor :math:`V_{pqrs}`\ .
+
+    **Example**
+
+    >>> symbols  = ['H', 'H']
+    >>> geometry = np.array([[0.0, 0.0, 0.0], [1.398397361, 0.0, 0.0]], requires_grad = False)
+    >>> mol = qml.qchem.Molecule(symbols, geometry)
+    >>> core, one, two = qml.qchem.electron_integrals(mol)()
+    >>> qml.qchem.chemist_transform(two, spatial_basis=True)
+    (tensor([[-0.427983, -0.      , -0.      , -0.      ],
+             [-0.      , -0.427983, -0.      , -0.      ],
+             [-0.      , -0.      , -0.439431, -0.      ],
+             [-0.      , -0.      , -0.      , -0.439431]], requires_grad=True),
+    tensor([[[[0.337378, 0.      ],
+             [0.       , 0.331856]],
+             [[0.      , 0.090605],
+             [0.090605 , 0.      ]]],
+            [[[0.      , 0.090605],
+             [0.090605 , 0.      ]],
+             [[0.331856, 0.      ],
+             [0.       , 0.348826]]]], requires_grad=True))
+
+    .. details::
+        :title: Theory
+
+        The two-electron integral in physicists' notation :math:`(\langle \cdot \rangle)` is defined as:
+
+        .. math::
+
+            \langle pq \vert rs \rangle = h_{pqrs} = \int \frac{\chi^*_{p}(x_1) \chi^*_{q}(x_2) \chi_{r}(x_1) \chi_{s}(x_2)}{|r_1 - r_2|} dx_1 dx_2,
+
+        while in chemists' notation :math:`([ \cdot ])` it is written as:
+
+        .. math::
+
+            [pq \vert rs] = V_{pqrs} = \int \frac{\chi^*_{p}(r_1) \chi^*_{q}(x_1) \chi_{r}(x_2) \chi_{s}(x_2)}{|r_1 - r_2|} dx_1 dx_2.
+
+        In the spin basis, this index reordering :math:`pqrs \rangle psrq` leads to formation of one-body terms :math:`h_{prrs}` that come out during
+        the coversion:
+
+        .. math::
+
+            :math:`h_{prrs}` = \int \frac{\chi^*_{p}(x_1) \chi^*_{r}(x_2) \chi_{r}(x_1) \chi_{s}(x_2)}{|x_1 - x_2|} dx_1 dx_2,
+
+        where :math:`\chi_{r}(x)` will have same spin, i.e., :math:`\chi_{r}(x_i) = \phi(r_i)\alpha(\omega)` or :math:`\chi_{r}(x_i) = \phi(r_i)\beta(\omega)`.
+
+    """
+
+    chemist_two_body_coeffs = np.swapaxes(two_body_tensor, 1, 3)
+    one_body_correction = -np.einsum("prrs", chemist_two_body_coeffs)
+
+    if spatial_basis:
+        chemist_two_body_coeffs = 0.5 * chemist_two_body_coeffs
+        one_body_correction = 0.5 * np.kron(one_body_correction, np.eye(2))
+
+    return one_body_correction, chemist_two_body_coeffs
+
+
+def build_spin_tensors(one_body_integrals, two_body_integrals=None):
```

Comment:
Is this function needed? If not, I suggest removing it.

Reply:
It is being used in the `basis_rotation` method now. Moreover, I think as we aim to extend our functionality to support fermionic operators, it would be a handy utility function.